### PR TITLE
Update msbuild for 2.1

### LINKED ISSUE #354: Make the msbuild repo build in dev/release/2.1
Get the msbuild repo to build both in the source-build repo and in the offline tarball build.

### PR DESCRIPTION
**Pull request by dseefeld**
Remove old patches and update msbuild command to build newer version of msbuild.  Fixes #354

### CODE REVIEW

File: repos/msbuild.proj [hunk 1]
```diff
@@ -2,25 +2,10 @@
 <Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))/dir.props" />
   <PropertyGroup>
-    <SolutionToBuild>$(ProjectDirectory)/NetCore.sln</SolutionToBuild>
-    <PackagesOutput>$(ProjectDirectory)/bin/$(Configuration)/packages</PackagesOutput>
-    <BuildCommand>$(DotnetToolCommand) pack $(SolutionToBuild) /p:Configuration=$(Configuration) /p:PackageOutputPath=$(PackagesOutput)</BuildCommand>
+    <PackagesOutput>$(ProjectDirectory)/artifacts/$(Configuration)/packages</PackagesOutput>
+    <BuildCommand>$(ProjectDirectory)build/build$(ShellExtension) build -DotNetBuildFromSource -DotNetCoreSdkDir $(DotNetCliToolDir) -bootstraponly -skiptests -pack -configuration $(Configuration) /p:DisableNerdbankVersioning=true /p:GitHeadSha=71eedf98975925fa7f2f3e669d19ba6c81391048</BuildCommand>
```
Comment (dagood) [inline repos/msbuild.proj]: I think the hard-coded commit hash should be stored in a property. (I'll have to do this for the auto-updater at some point anyway.) Unfortunately, we aren't standardized on `LatestCommit` or `GitCommitHash`, but more use `LatestCommit`.

Comment (dseefeld) [reply repos/msbuild.proj]: Agreed.  I thought it would be done as part of the updater task.

Comment (dagood) [reply repos/msbuild.proj]: Sounds good, just hard to spot at the end of a long line.

Comment (dseefeld) [reply repos/msbuild.proj]: I updated it so it doesn't get missed.

File: .gitmodules [hunk 2]
```diff
@@ -33,7 +33,7 @@
 	url = https://github.com/aspnet/websdk
 [submodule "src/msbuild"]
 	path = src/msbuild
-	url = https://github.com/Microsoft/msbuild
+	url = https://github.com/cdmihai/msbuild
```
Comment (cdmihai) [inline .gitmodules]: Master does not yet support the sourcebuild parameters, I was thinking to send that PR out when sourcebuild approaches master, but I guess I can just open the PR to our master now.

On another note, master does not work CLIs older than 2.1.3xx, should you set the gitmodule to point to https://github.com/Microsoft/msbuild/tree/sourceBuild_2.1.101?

File: repos/msbuild.proj [hunk 3]
```diff
@@ -2,25 +2,10 @@
 <Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))/dir.props" />
   <PropertyGroup>
-    <SolutionToBuild>$(ProjectDirectory)/NetCore.sln</SolutionToBuild>
-    <PackagesOutput>$(ProjectDirectory)/bin/$(Configuration)/packages</PackagesOutput>
-    <BuildCommand>$(DotnetToolCommand) pack $(SolutionToBuild) /p:Configuration=$(Configuration) /p:PackageOutputPath=$(PackagesOutput)</BuildCommand>
+    <PackagesOutput>$(ProjectDirectory)/artifacts/$(Configuration)/packages</PackagesOutput>
+    <BuildCommand>$(ProjectDirectory)build$(ShellExtension) build -DotNetBuildFromSource -DotNetCoreSdkDir $(DotNetCliToolDir) -bootstraponly -skiptests -pack -configuration $(Configuration) /p:GitHeadSha=1a685bea87d005f2ebcdf9a690012592e381e658</BuildCommand>
```
Comment (cdmihai) [inline repos/msbuild.proj]: Our windows build does not support sourcebuild, but `build$(ShellExtension)` suggests is should? Or is it just future proofing? :)

**Pull request merged**